spec/c14n_spec.rb

```diff
@@ -164,6 +164,10 @@
             canonicalize!(allow_query: :all).
             to_s.should eql('http://example.com/path?%ED=view')
         end
+
+        it 'does not error when there are bad things in query keys when allow_query isn\'t :all' do
+          BLURI('http://some.com/a/path?%E2').canonicalize!
```

Comment:
We should say that it doesn't raise an error at all (via `expect { thing }.not_to raise_error`) – yes, `EncodingError` is the one we're expecting in this case, but this spec should also fail if, say, an `ArgumentError` happens. There's an interesting discussion on the potential for false positives via `expect { thing }.not_to raise_error(SpecificError)` at this [RSpec issue](https://github.com/rspec/rspec-expectations/issues/231).